src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -148,18 +164,62 @@ class FilterButton extends React.Component<Props, State> {
         }
     };
 
-    deleteCondition = (index: number) => {
-        const { conditions } = this.state;
+    updateSelectedPrefix = (option: SelectOptionType) => {
+        const prefix = option.displayText;
 
-        const updatedConditions = conditions.filter((condition, conditionIndex) => {
-            return conditionIndex !== index;
+        this.setState({
+            selectedPrefix: prefix,
+        });
+
+        this.updateConditionsPrefixes(prefix);
+    };
+
+    updateConditionsPrefixes = (prefix: string) => {
+        const { conditions } = this.state;
+        const conditionsAfterDeletion = conditions.map((condition, index) => {
```

Comment:
is this actually the `conditionsAfterDeletion`? looks out of place in this function